roles/kubernetes/control-plane/defaults/main/main.yml

```diff
@@ -140,6 +140,50 @@ kube_api_runtime_config: []
 kube_token_auth: false
 kube_oidc_auth: false
 
+## Structured authentication config
+## Structured AuthenticationConfiguration is a new feature in Kubernetes v1.29+ (GA in v1.32) that configures the API server's authentication modes with a structured configuration file.
+## Note: The `--authentication-config` and `--oidc-*` flags are mutually exclusive. The two features cannot be used at the same time.
+## Docs: https://kubernetes.io/docs/reference/access-authn-authz/authentication/#using-authentication-configuration
+## Examples: https://kubernetes.io/blog/2024/04/25/structured-authentication-moves-to-beta/#how-to-use-structured-authentication-configuration
+## KEP: https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/3331-structured-authentication-configuration
+kube_apiserver_use_authentication_config_file: false
+kube_apiserver_authentication_jwt_provider: []
```

Comment:
I would just feed the whole config IMO.
It's quite possible other keys that `jwt` are added in the feature.
It would also make it easier to validate the whole stuff if k/k start publishing a schema.
We can merge the obvious default (apiVersion/kind/) at the use site